website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+
+  const findSelectedPluginIndex = useCallback(() => {
+    return plugins.findIndex((plugin) => plugin.enabled);
+  }, [plugins]);
+
+  const handlePluginSelect = useCallback(
+    (index: number) => {
+      const selectedIndex = plugins.findIndex((plugin) => plugin.enabled);
+      const newPlugins = plugins.map((plugin, i) => ({
+        ...plugin,
+        enabled: selectedIndex === index ? false : i === index,
+      }));
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      // onPluginActivated(newPlugins[index].enabled);
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginEdit = useCallback(
+    (index: number) => {
+      setSelectedPluginIndex(index);
+      onOpen();
+    },
+    [onOpen]
+  );
+
+  const handlePluginRemove = useCallback(
+    (index) => {
+      const updatedPlugins = plugins.filter((_, idx) => idx !== index);
+      setValue("plugins", updatedPlugins, { shouldDirty: true });
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginSave = useCallback(async () => {
+    if (textareaRef.current) {
+      const newPlugins = [...plugins];
+      newPlugins[selectedPluginIndex!].url = textareaRef.current.value;
+
+      const plugin: PluginEntry = await post(API_ROUTES.GET_PLUGIN_CONFIG, {
+        arg: { plugin: newPlugins[selectedPluginIndex!] },
+      });
+      // disable all existing plugins and enable new one
+      newPlugins.forEach((plugin) => (plugin.enabled = false));
+      newPlugins[selectedPluginIndex!].enabled = true;
+      newPlugins[selectedPluginIndex!].plugin_config = plugin.plugin_config;
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      //onPluginActivated(newPlugins[selectedPluginIndex!].enabled);
+    }
+    onClose();
+  }, [onClose, control, plugins, selectedPluginIndex]);
+
+  const handlePluginAdd = useCallback(() => {
+    const newPlugin: PluginEntry = {
+      url: "",
+      enabled: false,
+    };
+    setValue("plugins", [...plugins, newPlugin]);
+    handlePluginEdit(plugins.length);
+  }, [setValue, plugins, handlePluginEdit]);
+
+  const limitPluginNameLength = (name: string) => {
+    if (!name) {
+      return "";
+    }
+    if (name.length > 15) {
+      return name.substring(0, 15) + "...";
+    }
+    return name;
+  };
+
+  return (
+    <FormControl mb="40px" ref={containerRef}>
+      <Box position="fixed" zIndex="1">
+        <Menu placement="auto">
+          <MenuButton as={Button} pr="3" rightIcon={<Cog />} w={measureDivWidth(containerRef.current) + 15} size="lg">
+            {plugins && plugins[findSelectedPluginIndex()]?.enabled ? (
+              <Box display="flex" justifyContent="flex-start" gap={2}>
+                <Image
+                  src={plugins[findSelectedPluginIndex()]?.plugin_config?.logo_url}
+                  alt="Plugins"
+                  width="25px"
+                  height="25px"
+                  marginRight="5px"
+                />
+                <Text mt="4px" fontSize="15px">
```

Comment:
please use one of `sm` `md` `lg` etc..., so that it is consistent across screens and devices

Reply:
All of the sm, md, lg are too large I tried it all.
What else I can try?

Reply:
I put "sm" now.

Reply:
`xs`